tests/async/test_download.py

```diff
@@ -349,3 +349,20 @@ async def test_should_delete_downloads_on_browser_gone(browser_factory, server):
     assert os.path.exists(path1) is False
     assert os.path.exists(path2) is False
     assert os.path.exists(os.path.join(path1, "..")) is False
+
+
+async def test_download_cancel_should_work(browser, server):
+    def handle_download(request):
```

Comment:
Consider using context managers in new test code

Reply:
I want to keep the tests similar to upstream tests, so its easier to work with them in the future.